lib/msal-core/src/UserAgentApplication.ts

```diff
@@ -1158,14 +1146,9 @@ protected getCachedTokenInternal(scopes : Array<string> , user: User): CacheResu
             acquireTokenUserKey = Constants.acquireTokenUser + Constants.resourceDelimeter  + Constants.no_user +Constants.resourceDelimeter +  authenticationRequest.state;
         }
 
-          if (Utils.isEmpty(this._cacheStorage.getItem(acquireTokenUserKey))) {
-          this._cacheStorage.setItem(acquireTokenUserKey, JSON.stringify(userObject));
-        }
-
+        this._cacheStorage.setItem(acquireTokenUserKey, JSON.stringify(userObject));
         const authorityKey = Constants.authority + Constants.resourceDelimeter + authenticationRequest.state;
-        if (Utils.isEmpty(this._cacheStorage.getItem(authorityKey))) {
-          this._cacheStorage.setItem(authorityKey, acquireTokenAuthority.CanonicalAuthority);
-        }
+        this._cacheStorage.setItem(authorityKey, acquireTokenAuthority.CanonicalAuthority);
```

Comment:
are we missing the storeAuthStateInCookie flag here? You are passing it on line 412.

Reply:
same comment as above.